test/unit_test/explore_test.cc

```diff
@@ -49,8 +49,11 @@ BOOST_AUTO_TEST_CASE(new_sample_pdf)
   the_pdf.push_back({2.f, 3.5f, 0.1f / 1.5f});
   the_pdf.push_back({3.5f, 4.5f, 0.8f / 1.0f});
   the_pdf.push_back({4.5f, 6.2f, 0.1f / 1.7f});
-  const auto str_pdf = to_string(the_pdf);
-  std::cout << str_pdf << std::endl;
+  const auto str_pdf = to_string(the_pdf);  // 2-3.5:0.0666667,3.5-4.5:0.8,4.5-6.2:0.0588235
+  // avoid float precision compare
+  BOOST_CHECK_EQUAL(str_pdf.find("2-3.5:"), 0);
+  BOOST_CHECK(str_pdf.find(",3.5-4.5:0.8,4.5-6.2:") > 0);
```

Comment:
This wont be resilient to float differences. Is that okay? Also if the to_string changes then this test will break

Reply:
well its all generated from right above, any suggestions on how to replace that print? or just delete or what

Reply:
You could pick out the bits of it and assert that segment exists. It's way more work and not really worth it. I just wanted to point it out. Let's not worry about it

Reply:
if it passes it stays, if not we can later revisit. It is already wrong printing random stuff.